### PR TITLE
[1346] Add support for a "Rich Text" widget in forms

### LINKED ISSUE #1346: Add support for a rich text edition widget
It will behave similarly to the existing Textarea widget, except that the frontend will interpret the text value as Markdown and provide a WYSIWYG UI to edit it (and send the result as Markdown source).

It will be supported by the core APIs, the Views DSL and the Form Description Editor.

### PR DESCRIPTION
**Pull request by pcdavid**
# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [x] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [x] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [x] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [x] Have the relevant issues been added to the pull request?
- [x] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [x] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/FormDescriptionEditorRepresentation.tsx [hunk 1]
```diff
@@ -322,6 +323,17 @@ export const FormDescriptionEditorRepresentation = ({
...
+            id="RichText"
+            data-testid="FormDescriptionEditor-RichText"
+            draggable="true"
+            className={classes.widgetKind}
+            onDragStart={handleDragStart}>
+            <TextFormatIcon />
+            <Typography variant="caption" gutterBottom>
+              Rich Text
+            </Typography>
+          </div>{' '}
```
Comment (AxelRICHARD) [inline packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/FormDescriptionEditorRepresentation.tsx]: Why this {' '} ?

File: packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/RichTextWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  selected: {
+    color: theme.palette.primary.main,
+  },
+  editorContainer: {
+    marginTop: theme.spacing(2),
+    borderRadius: '2px',
+    borderColor: theme.palette.divider,
+    borderWidth: '1px',
+    borderStyle: 'solid',
+    color: '#000',
```
Comment (AxelRICHARD) [inline packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/RichTextWidget.tsx]: Are you sure about this color value ?

File: packages/view/backend/sirius-components-view-edit/plugin.properties [hunk 3]
```diff
@@ -0,0 +1,283 @@
+# Copyright (c) 2021, 2022 Obeo.
```
Comment (AxelRICHARD) [inline packages/view/backend/sirius-components-view-edit/plugin.properties]: Why this plugin.properties file is only committed now?

**Review by AxelRICHARD — CHANGES_REQUESTED** [review]

File: packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/FormDescriptionEditorRepresentation.tsx (hunk 1 above)
Comment (pcdavid) [reply packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/FormDescriptionEditorRepresentation.tsx]: Done.

File: packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/RichTextWidget.tsx (hunk 2 above)
Comment (pcdavid) [reply packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/RichTextWidget.tsx]: The black `color`? What's the issue with it?

Comment (AxelRICHARD) [reply packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/RichTextWidget.tsx]: None, I just wanted to be sure that you want to put here a specific color, that do not comes from the sirius color theme.

Comment (pcdavid) [reply packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/RichTextWidget.tsx]: Fixed to use `theme.palette.text.primary`.

File: packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/components/RichTextComponentProps.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Obeo.
```
Comment (AxelRICHARD) [inline packages/forms/backend/sirius-components-forms/src/main/java/org/eclipse/sirius/components/forms/components/RichTextComponentProps.java]: 2020 -> 2022

File: packages/forms/frontend/sirius-components-forms/src/richtexteditor/RichTextEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+    editorState: () => $convertFromMarkdownString(value, TRANSFORMERS),
+  };
+
+  return (
+    <LexicalComposer initialConfig={initialConfig}>
+      <ToolbarPlugin readOnly={readOnly} onSave={onSave} />
+      <div className={classes.editorContainer}>
+        <RichTextPlugin
+          contentEditable={<ContentEditable onFocus={onFocus} readOnly={readOnly} />}
+          placeholder={<div className={classes.editorPlaceholder}>{placeholder}</div>}
```
Comment (AxelRICHARD) [inline packages/forms/frontend/sirius-components-forms/src/richtexteditor/RichTextEditor.tsx:256]: The placeholder has been put in a weird position, under text text area:
Capture d’écran 2022-11-07 à 10 35 26
Is it possible to put on the text area?

**Review by AxelRICHARD — CHANGES_REQUESTED** [review]

File: packages/view/backend/sirius-components-view-edit/src/main/java/org/eclipse/sirius/components/view/provider/ViewItemProviderAdapterFactory.java [hunk 6]
```diff
@@ -821,6 +821,29 @@ public Adapter createTextAreaDescriptionAdapter() {
         return this.textAreaDescriptionItemProvider;
     }
 
+    /**
```
Comment (AxelRICHARD) [inline packages/view/backend/sirius-components-view-edit/src/main/java/org/eclipse/sirius/components/view/provider/ViewItemProviderAdapterFactory.java:824]: Please add new RichTextDescription in org.eclipse.sirius.components.view.provider.FormDescriptionItemProvider.collectNewChildDescriptors(Collection<Object>, Object)

**Review by AxelRICHARD — CHANGES_REQUESTED** [review]

**Review by AxelRICHARD — APPROVED** [review]

**Pull request merged**